libr/anal/p/anal_arm_cs.c

```diff
@@ -1572,9 +1578,23 @@ static int analop_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len
 		MATH32_NEG("|");
 		break;
 	case ARM_INS_LSR:
+	        if (insn->detail->arm.update_flags) {
+			if (OPCOUNT() == 2) {
+				r_strbuf_appendf (&op->esil, "%s,!,!,?{,%s,1,%s,-,0x1,<<,&,!,!,cf,:=,},",ARG(1),ARG(0),ARG(1));
```

Comment:
is the trailing comma needed here?

Reply:
yes, as other flags need to be updated afterwards

lsrs r1, r2         r2,!,!,?{,r1,1,r2,-,0x1,<<,&,!,!,cf,:=,},r2,r1,>>,0xffffffff,&,r1,=,$z,zf,:=,31,$s,nf,:=      

lsrs update the N and Z flags according to the result